### ISSUE TITLE
Bug: (es5 multine string with closing html tag)

### ISSUE THREAD

**Issue by zx-81-16k** [eslint/eslint#15116]
### Environment

Node version: v12.22.5
npm version: v6.14.14
Local ESLint version: v7.14.0
Global ESLint version: Not found
Operating System: Windows 7

### What parser are you using?

vue-eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports =
{
    'env':
    {
        'browser': true,
        'es6': true
    },
    'extends':
    [
        'plugin:vue/recommended',
        'eslint:recommended'
    ],
    'globals':
    {
        'Atomics': 'readonly',
        'SharedArrayBuffer': 'readonly'
    },
    'parserOptions':
    {
        'ecmaVersion': 2018,
        'sourceType': 'module'
    },
    'plugins':
    [
        'vue'
    ],
    "ignorePatterns": ["/static/*"],
    'rules':
    {
		'vue/valid-v-model': ['off'],
		'vue/no-v-html': ['off'],
        'vue/html-indent': ['off', 'tab'],
        'vue/html-closing-bracket-spacing': ['off'],
        'vue/max-attributes-per-line': ['off'],
        'vue/html-quotes': ['warn', 'single'],
        'vue/no-unused-vars': ['warn'],
        'vue/return-in-computed-property': ['warn'],
        'vue/require-v-for-key': ['off'],
        'no-undef': ['off'],
        'no-unused-labels': ['warn'],
        'no-unused-vars': ['warn'],
		'no-mixed-spaces-and-tabs': ['off'],
		'no-unreachable': ['off']
    }
}

```
</details>

```js
		head(){
			return " \
				<!-- Google Tag Manager --> \
				<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': \
				new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0], \
				j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src= \
				'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f); \
				})(window,document,'script','dataLayer','GTM-NHXQ54L'); </script> \
				<!-- End Google Tag Manager --> \
				"								
		}
```

### What did you expect to happen?

no warning, no errors.

### What actually happened?

at end of this line i get "unterminated string"
`})(window,document,'script','dataLayer','GTM-NHXQ54L'); </script> \`
if i escape '/'
`})(window,document,'script','dataLayer','GTM-NHXQ54L'); <\/script> \`
then i get "unncecsary escape" 

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ljharb** [comment]
In browsers when inline in an HTML document, any occurrence of `</script>`, even in a string, immediately ends the script tag - this probably isn't the issue here but I thought I'd mention it.

**Comment by nzakas** [comment]
This issue doesn't occur with the default ESLint parser, and so I'd suggest following up with the `vue-eslint-parser` project for help.
